stdlib/public/Darwin/Foundation/NSDictionary.swift

```diff
@@ -107,19 +107,84 @@ extension Dictionary : _ObjectiveCBridgeable {
     result = builder.take()
   }
 
+  private static func _typePun<T>(index:Int, of buffer: UnsafeBufferPointer<AnyObject>, to: T.Type) -> UnsafeMutablePointer<T> {
+    let typedBase = UnsafeMutableRawPointer(mutating: buffer.baseAddress!).assumingMemoryBound(to: T.self)
+    let unboundTarget = UnsafeMutableRawPointer(mutating: typedBase + index)
+    return unboundTarget.bindMemory(to: T.self, capacity: 1)
+  }
+
+  private static func _verbatimBridge<T>(
+    _ objectBuffer: UnsafeBufferPointer<AnyObject>,
+    to: T.Type
+  ) {
+    for i in (0..<objectBuffer.count).reversed() {
```

Comment:
This has to go in reverse because if sizeof(T) > sizeof(AnyObject), we'll be shifting each element as we bridge it